### PR TITLE
Fix some CI tasks

### PR DESCRIPTION
**Pull request by mariomac**
It excludes generated files from the coverage reports, so the percentages are more realistic.

It also fixes a typo in the push_image.yml github action.

It also fixes a flaky test.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -92,22 +92,22 @@ compile:
 .PHONY: test
 test:
 	@echo "### Testing code"
-	GOOS=$(GOOS) go test -mod vendor -a ./... -coverpkg=./... -coverprofile cover.out
+	GOOS=$(GOOS) go test -mod vendor -a ./... -coverpkg=./... -coverprofile cover.all.out
```
Comment (OlivierCazade) [inline Makefile:95]: Did you try to adapt the coverpkg option here?

In theory we should be able to exclude unwanted files here and directly have a clean output.

**Review by OlivierCazade — APPROVED** [review]
LGTM since my comment is related to a mechanism not modified by this PR.

File: Makefile (hunk 1 above)
Comment (mariomac) [reply Makefile:95]: Yeah I tried, but `coverpkg` only allows you picking up a single package and not a bunch of packages, nor just excluding selected packages.

Comment (OlivierCazade) [reply Makefile:95]: Ok, thanks!

**Pull request merged**